File: rb/features.rb
# EuVatValidation SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module EuVatValidationFeatures
  def self.make_feature(name)
    case name
    when "base"
      EuVatValidationBaseFeature.new
    when "ratelimit"
      EuVatValidationRatelimitFeature.new
    when "retry"
      EuVatValidationRetryFeature.new
    when "test"
      EuVatValidationTestFeature.new
    when "timeout"
      EuVatValidationTimeoutFeature.new
    else
      EuVatValidationBaseFeature.new
    end
  end
end

File: py/features.py
# EuVatValidation SDK feature factory

from feature.base_feature import EuVatValidationBaseFeature
from feature.test_feature import EuVatValidationTestFeature


def _make_feature(name):
    features = {
        "base": lambda: EuVatValidationBaseFeature(),
        "test": lambda: EuVatValidationTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
